### ISSUE TITLE
Should `autocov` auto-broadcast over matrices?

### ISSUE THREAD

**Issue by ParadaCarleton** [JuliaStats/StatsBase.jl#779]
I ask this because the current default behavior of `autocov` is that, if provided with a matrix, it broadcasts over the columns of the matrix.

I'm not sure this is a good idea. If a user wants to apply `autocov` columnwise, they can use `mapslices` or `eachcol`. There's no particular need to broadcast automatically for them. Auto-broadcasting is generally [considered bad form](https://discourse.julialang.org/t/why-are-there-broadcast-versions-of-functions/4708/6), as it takes the choice of whether or not to broadcast away from the user.

However, broadcasting `autocov` columnwise has several drawbacks/ambiguities:
1. A matrix can easily be used to represent panel data. However, panel autocorrelations are typically estimated differently than by just estimating the autocovariance for each lag, as this estimator is both inefficient and inconsistent.
2. If the series are different features, the series-wise covariance estimator is still inefficient if the cross-correlations are not zero.
3. It's not obvious whether this method should return just the covariances at each lag or a full matrix-of-matrices containing cross-covariances as well.
4. The columns/rows of a matrix are easy to interchange or confuse, and a user not paying close attention to the documentation might mistakenly believe the function is broadcasted rowwise.

As a result, I propose that broadcasting `autocov` over matrices be deprecated and eventually removed.

**Comment by nalimilan** [comment]
Yes it's not great, but this behavior is inherited/consistent with `cor` and `cov` in Statistics. We will probably be able to deprecate these forms once Statistics moves out of the stdlib and is merged with StatsBase.

What would the replacement syntax look like exactly? One concern could be that computing the result for several lags would be much more verbose than currently.

**Comment by devmotion** [comment]
> The columns/rows of a matrix are easy to interchange or confuse, and a user not paying close attention to the documentation might mistakenly believe the function is broadcasted rowwise.

https://github.com/JuliaLang/julia/pull/32310 is supposed to solve this issue.
